analytic_diffuse/integrators.py

```diff
@@ -0,0 +1,55 @@
+from scipy import integrate
+import numpy as np
+from . import models
+from scipy.special import j0, jn_zeros
+import mpmath as mp
+
+def hankel_solver(model: [callable, str], u: [float, np.ndarray], quad_kwargs={}, high_precision=False, use_points=False, *args, **kwargs):
+    if callable(model):
+        assert model.__name__ in models._funcnames
+    else:
+        model = getattr(models, model)
+
+    if model.__name__ in models._projected_funcs:
+        # using phi here because that's what we use in the paper, though in models.py it's theta
```

Comment:
Everything should have phi as azimuth and theta as co-latitude. Where in models.py is it otherwise?